### ISSUE TITLE
Action methods cannot be overridedden

### ISSUE THREAD

**Issue by pvasek** [mobxjs/mobx#379]
It seems the code (typescript) below was working before PR #313 and now it's broken.

```typescript
class FormModel {
  @action save() {
  ...
  }
}

class MyForm extends FormModel {
  @action save() {
     ...
    super();
  }
}
```

This code doesn't work either:

```typescript
class FormModel {
  @action save() {
  ...
  }
}

class MyForm extends FormModel {
  save() {
     ...
    super();
  }
}
```

Of course this workaround works fine:
```typescript
class FormModel {
  protected internalSave() {
   ...
  }
  @action save() {
      this.internalSave();
  }
}

class MyForm extends FormModel {
  internalSave() {
     ...
    super();
  }
}
```

I am not a big fun of inheritance in many cases but in this one it is really helpful. We can have a general form model which then can be extended for particular forms.

Is the inheritance something that can be used with mobx and this is a bug or you see it as evil and don't want to support it?

**Comment by mweststrate** [comment]
I'm not a huge, fan of inheritance indeed, but there are cases where it does fit and I think MobX shouldn't be opinionated about that. I released `2.3.4` earlier to day which fixes this issue :) Happy inheriting!

**Comment by pvasek** [comment]
That's perfect. Thank you!

**Comment by mciastek** [comment]
@mweststrate I think this one is still valid, because I can't use inheritance with `@action.bound`. When I try to extend inherited method, I get `[mobx] @action fields are not reassignable`.

**Comment by mweststrate** [comment]
@mciastek this issue is closed for 2.5 years. Please open a new one.
